src/main/java/bool/variant/cnf/interpreter/decisionGraph/DecisionGraph.java

```diff
@@ -0,0 +1,226 @@
+    }
+
+    /**
+     * Get the conflict Clause to add for the next run AND
+     * Get all variables that have been backtracked
+     */
+    public Clause constructConflictClause(Map<Variable, Boolean> assignment) {
+        Conflict conflict = findConflict();
+        Decision firstUIP = getFirstUIP(conflict);
+        Logger.info(1, "First UIP: " + firstUIP.decision);
+        Clause conflictClause = conflict.getConflictClause();
+        Logger.info(2, "Conflict Ancestors:\n" + conflictClause.stream()
+                .map(AbstractVariable::negated)
+                .map(this::findDecision)
+                .filter(x -> x instanceof HasAncestors)
+                .map(x -> (HasAncestors) x)
+                .map(x -> x.getAllAncestorsOnSameLevel().stream()
+                        .map(anc -> anc.decision.toString())
+                        .collect(Collectors.joining(", ", "\t\t", "")))
+                .collect(Collectors.joining("\n")));
+        Logger.info(2, "Unexpanded conflict clause: " + conflictClause);
+        List<Decision> expandedConflictParticipants = new ArrayList<>();
+
+        for(Decision decision : conflict.ancestors)
+            if(decision instanceof ForcedDecision fd)
+                conflictClause = expandConflictClauseToFirstUIP(fd, conflictClause, firstUIP, expandedConflictParticipants);
+
+        Logger.info(1, "Expanded conflict clause: " + conflictClause);
+
+        // minimize
+        int oldConflictClauseSize;
+        do {
+            oldConflictClauseSize = conflictClause.size();
+            Clause finalResultingConflictClause = conflictClause;
+            conflictClause = new Clause(
+                    conflictClause.stream()
+                            .filter(var -> !(findDecision(var) instanceof ForcedDecision fd) ||
+                                    !finalResultingConflictClause.containsAllVariables(
+                                            fd.getReasons().stream()
+                                                    .map(AbstractVariable::getVariable)
+                                                    .toList()))
+                            .toList()
+            );
+        } while(conflictClause.size() < oldConflictClauseSize);
+
+        // decrease level until conflict clause is satisfiable
+        Map<Variable, Boolean> currentAssignment = new HashMap<>(assignment);
+        while(level > 0 && conflictClause.getStatus(currentAssignment) == Clause.Status.UNSAT) {
+            stream()
+                    .filter(node -> node.level == level)
+                    .filter(node -> node instanceof Decision)
+                    .map(node -> (Decision) node)
+                    .filter(decision -> currentAssignment.containsKey(decision.decision.getVariable()))
+                    .forEach(decision -> currentAssignment.remove(decision.decision.getVariable()));
+            level--;
+        }
+
+        conflictClause.resetWatcherIndices();
```

Comment:
This should be inside of the loop, since a satisfiable assignment could be overread with the wrong watchers